internal/upstream_process.go

```diff
@@ -15,9 +15,11 @@ type UpstreamProcess struct {
 }
 
 func NewUpstreamProcess(name string, arg ...string) *UpstreamProcess {
+	cmd := exec.Command(name, arg...)
+	cmd.Stdin = os.Stdin
```

Comment:
We currently connect `stdout` and `stderr` [inside `Run()`](https://github.com/basecamp/thruster/blob/main/internal/upstream_process.go#L25-L26). It would be good to handle these all in the same place, for consistency.

Maybe move the `stdin` case to `Run()` too, to match the others?

Reply:
Excellent, thanks again for diagnosing and fixing! 🎉